pkg/ocm/ocm.go

```diff
@@ -0,0 +1,161 @@
+package ocm
+
+import (
+	"context"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/openshift/insights-operator/pkg/config"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/klog/v2"
+)
+
+const (
+	targetNamespaceName = "openshift-config-managed"
+	secretName          = "simple-content-access-cert"
+)
+
+// Controller holds all the required resources to be able to communicate with OCM API
+type Controller struct {
+	coreClient   corev1client.CoreV1Interface
+	context      context.Context
+	configurator Configurator
+}
+
+// Configurator represents the interface to retrieve the configuration for the gatherer
+type Configurator interface {
+	Config() *config.Controller
+	ConfigChanged() (<-chan struct{}, func())
+}
+
+// New creates new instance
+func New(ctx context.Context, coreClient corev1client.CoreV1Interface, configurator Configurator) *Controller {
+	return &Controller{
+		coreClient:   coreClient,
+		context:      ctx,
+		configurator: configurator,
+	}
+}
+
+// Run periodically queries the OCM API and update corresponding secret accordingly
+func (c *Controller) Run() {
+	cfg := c.configurator.Config()
+	endpoint := cfg.OcmEndpoint
+	interval := cfg.OcmInterval
+	configCh, cancel := c.configurator.ConfigChanged()
+	defer cancel()
+
+	c.requestDataAndCheckSecret(endpoint)
+	for {
+		select {
+		case <-time.After(interval):
+			c.requestDataAndCheckSecret(endpoint)
+		case <-configCh:
+			cfg := c.configurator.Config()
+			interval = cfg.OcmInterval
+			endpoint = cfg.OcmEndpoint
+		}
+	}
+}
+
+func (c *Controller) requestDataAndCheckSecret(endpoint string) {
+	data, err := requestData(endpoint)
+	if err != nil {
+		klog.Errorf("errror requesting data from %s: %v", endpoint, err)
```

Comment:
Don't we need a return here?

Reply:
Yeah probably yeah. Good catch..the error states will need an update anyway. I think we will introduce new IO state (something like `ScaDataDegraded`) and then probably a new alert based on this state.